lib/ripper-tags/data_reader.rb

```diff
@@ -130,20 +132,25 @@ def normalize_encoding(str)
 
     def each_tag
       return to_enum(__method__) unless block_given?
+
       file_finder.each_file do |file|
         begin
           $stderr.puts "Parsing file #{file}" if options.verbose
           extractor = tag_extractor(file)
         rescue => err
-          if options.force
-            $stderr.puts "Error parsing `#{file}': #{err.message}"
-          else
+          # print the error
+          $stderr.puts "Error parsing `#{file}': #{err.message}"
+          @error_count += 1
+
+          # Die if we ran without --force and we haven't seen any tags yet.
+          if !options.force && @tag_count == 0
```

Comment:
One final thought about this logic: suppose you run `ripper-tags on a `lib/` directory which has 2 files in it: `good.rb` and `erb_template.rb`. The first one will generate some tags, but the second one will enounter a parse error. Because there are already some tags, this condition will *not* halt processing, but just print to stderr and eventually return with exit status 2.

However, let's say that `erb_template.rb` gets processed before `good.rb`. Now the processing halts immediately because the first file had a parse error before any tags were able to get generated. The behavior of ripper-tags is suddenly very different depending on the order of input files.

I know that checking the number of tags is something that I've suggested initially, but now I don't think it's such a great idea. Let's pick consistency and simplicity. How about forgiving all parse errors (while outputting them to stderr) regardless of the number of tags being collected at the time of encountering the error? Let me know your thoughts.